### ISSUE TITLE
Tendenci to Support Multi-Factor Authentification

### ISSUE THREAD

**Issue by ndavis2** [tendenci/tendenci#830]
Would like to add an extra layer of security by adding multi-factor-authentication to Tendenci

**Comment by AustinPowered** [comment]
Plus one for adding support for MFA. Ideally, any user would be able to add MFA to their login. The ability to optionally require certain groups to use MFA would be a plus.

I didn't realize multi-factor authentication (MFA) wasn't already included in Django with django.contrib.auth. However, implementation of this feature is well documented. A quick search finds [django-two-factor-auth](https://pypi.org/project/django-two-factor-auth/) from PyPi. There is a [Stack Overflow post](https://stackoverflow.com/questions/54908541/django-two-factor-authentication) that mentions the [pyOTP library](https://github.com/pyauth/pyotp) for generating and verifying one-time passwords. pyOTP works with Google Authenticator and other RFC 4226 compatible apps.

Does anyone have experience with implementing this feature within Tendenci or another app?

Thanks,
AustinPowered

**Comment by goetzk** [comment]
I've done two MFA implementations for Django/Tendenci - seems only one was made public - though you will notice not recently...
https://github.com/goetzk/django-otp-vip 

I also did work with classic 2fa and saml vai these libraries. https://github.com/goetzk/django-two-factor-auth https://github.com/goetzk/djangosaml2

tl;dr: they can be integrated .
